src/test/java/com/connexta/search/index/IndexServiceImplTest.java

```diff
@@ -6,239 +6,163 @@
-import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.core.io.Resource;
 import org.springframework.http.HttpStatus;
-import org.springframework.web.reactive.function.client.WebClient;
-import org.springframework.web.reactive.function.client.WebClient.RequestHeadersSpec;
-import org.springframework.web.reactive.function.client.WebClient.RequestHeadersUriSpec;
-import org.springframework.web.reactive.function.client.WebClient.ResponseSpec;
-import reactor.core.publisher.Mono;
 
 @ExtendWith(MockitoExtension.class)
 class IndexServiceImplTest {
 
-  @Mock private SearchManager mockSearchManager;
+  static final String DATASET_ID = "00067360b70e4acfab561fe593ad3f7a";
+  static final String FILE_URL = "http://file";
+  static final String IRM_URL = "http://irm";
 
-  @AfterEach
-  public void afterEach() {
-    verifyNoMoreInteractions(mockSearchManager);
+  final ContentExtractorImpl mockContentExtractor = mock(ContentExtractorImpl.class);
+  final IndexRepository mockIndexRepository = mock(IndexRepository.class);
+  final IonResourceLoader mockIonResourceLoader = mock(IonResourceLoader.class);
+
+  final IndexService indexService =
+      new IndexServiceImpl(mockIndexRepository, mockIonResourceLoader, mockContentExtractor);
+  Index testIndex;
+
+  @BeforeEach
+  void beforeEach() {
+    testIndex = Index.builder().id(DATASET_ID).fileUrl(FILE_URL).irmUrl(IRM_URL).build();
   }
 
-  @ParameterizedTest
-  @ValueSource(ints = {400, 401, 403, 500, 501})
-  void testIrmUriReturnsNotOkStatusCode(final int code) throws Exception {
-    final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+  @AfterEach
+  void afterEach() {
+    reset(mockContentExtractor, mockIndexRepository, mockIonResourceLoader);
+  }
 
-    final SearchException thrown;
-    try (final MockWebServer storeMockWebServer = new MockWebServer()) {
-      storeMockWebServer.start();
-      storeMockWebServer.enqueue(new MockResponse().setResponseCode(code));
-
-      thrown =
-          assertThrows(
-              SearchException.class,
-              () ->
-                  ((IndexService) new IndexServiceImpl(mockSearchManager, WebClient.create()))
-                      .index(
-                          datasetId,
-                          storeMockWebServer
-                              .url(String.format("/dataset/%s/irm", datasetId))
-                              .uri()));
-    }
-    assertThat(thrown.getStatus(), is(HttpStatus.BAD_REQUEST));
+  @Test
+  void testIndex() throws IOException, TikaException {
+    doReturn(IOUtils.toInputStream("asdf", StandardCharsets.UTF_8))
+        .when(mockIonResourceLoader)
+        .get(anyString());
+    doReturn("fileContents").when(mockContentExtractor).extractText(any(InputStream.class));
+    indexService.index(testIndex);
+    verify(mockIndexRepository).save(testIndex);
```

Comment:
❗️This should verify that the final index has the values that you expect, not just that is the same object that was originally passed in. This test should verify that the contents is populated with the expected value in the index that is saved.